### PR TITLE
[SHIRO-842] log4j update

### PR DESCRIPTION
**Pull request by bmarwell**
- Fixes SHIRO-842
- removes all log4j1 usages (test usages)
- replaces "test if logged" with LoggerContextRule

TBD:
- [x] Need to remove the code cleanup from my IDE
- [x] Replace all `log4j.properties` with `log4j2.xml`

Following this checklist to help us incorporate your contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SHIRO) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[SHIRO-XXX] - Fixes bug in SessionManager`,
       where you replace `SHIRO-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean install apache-rat:check` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [X] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 
Trivial changes like typos do not require a JIRA issue (javadoc, comments...). 
In this case, just format the pull request title like `(DOC) - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [X] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -136,13 +136,19 @@
...
         </dependency>
         <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-log4j12</artifactId>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-slf4j-impl</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
```
Comment (bdemers) [inline core/pom.xml:144]: nit: scope these with `test`

I know they are scoped in the parent's dep management, but it makes it easier to debug.

File: core/src/test/java/org/apache/shiro/authc/AbstractAuthenticatorTest.java [hunk 2]
```diff
@@ -184,11 +182,10 @@ protected AuthenticationInfo doAuthenticate(AuthenticationToken token) throws Au
         }catch(AuthenticationException expectedException){
         }
 
-        String logMsg = out.toString();
+        final ListAppender listAppender = loggerContextRule.getListAppender("List");
+        String logMsg = String.join("\n", listAppender.getMessages());
```
Comment (bdemers) [inline core/src/test/java/org/apache/shiro/authc/AbstractAuthenticatorTest.java:186]: Wow! that log4j rule really cleaned this logic up! Nice find!

**Review by bdemers — APPROVED** [review]
+1
Feel free to ignore my dependency scope comment, that's something we can clean up in bulk after (and include all of the other test dependencies at the same time)

File: core/pom.xml (hunk 1 above)
Comment (bmarwell) [reply core/pom.xml]: Done

**Review by fpapon — APPROVED** [review]

**Pull request merged**